src/parser/shared/modules/HotTracker.ts

```diff
@@ -18,27 +28,25 @@ const extensionDebug = false; // logs pertaining to extensions
 const applyRemoveDebug = false; // logs tracking HoT apply / refresh / remove
 const healDebug = false; // logs tracking HoT heals
 
-class HotTracker extends Analyzer {
+abstract class HotTracker extends Analyzer {
```

Comment:
This whole class is a mess and really needs a refactor (or just replacement). A lot of its features are currently unused. I intend to do work on it in the future, but for now I preserved as much of it as I could and relearned a lot of what was going on in the process of adding types.